xarray/backends/zarr.py

```diff
@@ -331,29 +313,34 @@ def get_variables(self):
                                  for k, v in self.ds.arrays())
 
     def get_attrs(self):
-        _, attributes = _get_zarr_dims_and_attrs(self.ds, _DIMENSION_KEY)
-        return _decode_zarr_attrs(attributes)
+        attributes = OrderedDict(self.ds.attrs.asdict())
+        return attributes
 
     def get_dimensions(self):
-        dimensions, _ = _get_zarr_dims_and_attrs(self.ds, _DIMENSION_KEY)
+        dimensions = OrderedDict()
+        for k, v in self.ds.arrays():
+            try:
+                for d, s in zip(v.attrs[_DIMENSION_KEY], v.shape):
+                    dimensions[d] = s
+            except KeyError:
+                raise KeyError("Zarr object is missing the attribute `%s`, "
+                               "which is required for xarray to determine "
+                               "variable dimensions." % (_DIMENSION_KEY))
```

Comment:
@rabernat - I want to get your take on this change. What I've done is remove the `_DIMENSION_KEY` attribute from the dataset all together and I'm just pulling it from the individual arrays/variables. I realized we were storing this information multiple places and it was getting too tricky keeping everything straight.

Reply:
I see your point that the coord information was stored twice.

My understanding was that having `_DIMENSION_KEY` in the global dataset attributes was necessary for `set_coords` to work on non-dimensions coordinates. But maybe this is wrong--I can't find any justification for it by looking through the code.

Reply:
I suppose one option here is to just have no dimensions if there is no `_DIMENSION_KEY` on the variable. That would allow us to open any zarr store, even those not created by xarray.

Reply:
>  My understanding was that having _DIMENSION_KEY in the global dataset attributes was necessary for set_coords to work on non-dimensions coordinates. But maybe this is wrong--I can't find any justification for it by looking through the code.

 Right, I think that may have been the case but I've worked around that now.

> I suppose one option here is to just have no dimensions if there is no _DIMENSION_KEY on the variable. That would allow us to open any zarr store, even those not created by xarray.

I think this is doable but I want to leave that for another PR since we'll need to make a number of decisions about how to name dimensions that were not specified in the store.

Reply:
👍